a3p-integration/proposals/g:ymax1/test/ymax0.test.js

```diff
@@ -0,0 +1,139 @@
+// @ts-check
+import '@endo/init/debug.js';
+
+import { LOCAL_CONFIG, makeVstorageKit } from '@agoric/client-utils';
+import {
+  getVatInfoFromID,
+  getDetailsMatchingVats,
+} from '@agoric/synthetic-chain';
+import anyTest from 'ava';
+import { walletUpdates } from '../walletUpdates.js';
+import { bundleId, ymax0ControlAddr as ymaxControlAddr } from './consts.js';
+import { redeemInvitation, submitYmaxControl } from '../ymax-util.js';
+import { makeActionId, sendWalletAction } from '../wallet-util.js';
+
+/**
+ * @import {BridgeAction} from '@agoric/smart-wallet/src/smartWallet';
+ */
+
+const { fromEntries } = Object;
+
+const contractName = 'ymax0';
+
+const privateArgsOverrides = harden({
+  gmpAddresses: {
+    AXELAR_GAS: 'axelar1gasnew',
+    AXELAR_GMP: 'axelar1gmpnew',
+  },
+});
+
+const vsc = makeVstorageKit({ fetch }, LOCAL_CONFIG);
+const wup = walletUpdates(
+  () => vsc.readPublished(`wallet.${ymaxControlAddr}`),
+  { setTimeout, log: () => {} },
+);
+
+/** @param {any} x */
+const boardId = x => x.getBoardId();
+
+const test =
+  /** @type {import('ava').TestFn<Awaited<ReturnType<typeof makeTestContext>>>} */ (
+    anyTest
+  );
+
+/** @param {import('ava').ExecutionContext} t */
+const makeTestContext = async t => {
+  const { [contractName]: instance } = fromEntries(
+    await vsc.readPublished(`agoricNames.instance`),
+  );
+
+  const potentialVats = await getDetailsMatchingVats(contractName).then(
+    candidates => candidates.filter(v => !v.terminated),
+  );
+  if (potentialVats.length !== 1) {
+    t.log('WARN', 'Multiple running vats found', potentialVats);
+  }
+
+  const vatDetails = potentialVats.slice(-1)[0];
+  t.log(vatDetails);
+
+  return {
+    instanceId: boardId(instance),
+    vatDetails,
+  };
+};
+
+test.before(async t => (t.context = await makeTestContext(t)));
+
+test.serial('null upgrade existing instance with args override', async t => {
+  const id = makeActionId('upgrade');
```

Comment:
how does this work? oh. `makeActionId` has ambient access to the clock.

Please don't make me read code like that :(

Reply:
Sorry, there is a lot of ambient powers in a3p, so I figured that wouldn't hurt too much.